src/ncp/ncp_host.cpp

```diff
@@ -71,6 +75,76 @@ void NcpHost::GetDeviceRole(DeviceRoleHandler aHandler)
     mNcpSpinel.GetDeviceRole(aHandler);
 }
 
+void NcpHost::Join(const otOperationalDatasetTlvs &aActiveOpDatasetTlvs, const AsyncResultReceiver aReceiver)
+{
+    ChainedAsyncResultPtr result;
+    bool                  isAlready = !mJoinReceivers.empty();
+
+    VerifyOrExit(!isAlready);
+
+    result = std::make_shared<ChainedAsyncResult>();
+    result
+        ->Then([this](otError aError, ChainedAsyncResultPtr aNext) {
+            if (aError != OT_ERROR_NONE)
+            {
+                CallAsyncReceivers(mJoinReceivers, aError, "Failed to set active dataset!");
+            }
+            else
+            {
+                mNcpSpinel.Ip6SetEnabled(true, aNext);
+            }
+        })
+        ->Then([this](otError aError, ChainedAsyncResultPtr aNext) {
+            if (aError != OT_ERROR_NONE)
+            {
+                CallAsyncReceivers(mJoinReceivers, aError, "Failed to enable the network interface!");
+            }
+            else
+            {
+                mNcpSpinel.ThreadSetEnabled(true, aNext);
+            }
+        })
+        ->Then([this](otError aError, ChainedAsyncResultPtr aNext) {
+            OTBR_UNUSED_VARIABLE(aNext);
+            CallAsyncReceivers(mJoinReceivers, aError,
+                               aError == OT_ERROR_NONE ? "Success" : "Failed to enable the Thread network!");
+        });
```

Comment:
The order of `Then` is reversed from the actual execution order. Is it possible to use the execution order?

Reply:
I misread this. Never mind.